### PR TITLE
Make cache durable and loaded at startup from disk

### LINKED ISSUE #1806: Ensure cache consistency when the service boots up for realm, resolvers and storages
Realms have an unicity constraint on the openIdConfig uri that cannot be enforced during the service bootstrap.
Resolvers have an unicity constraint on the priority that cannot be enforced during the service bootstrap.

Replay the current events by tag for these resources when the service starts to populate the cache and bind the routes only after this operation is complete.

### PR DESCRIPTION
**Pull request by imsdu**
Fixes #1806

### CODE REVIEW

File: delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala [hunk 1]
```diff
@@ -180,8 +180,10 @@ final class ResolversImpl(
 
 object ResolversImpl {
 
+  final private[resolvers] case class ResolverKey(project: ProjectRef, iri: Iri)
```
Comment (umbreak) [inline delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala:183]: This seems to be unrelated to the current PR, isn't it?

I would suggest any improvement to the current code to be made in a separate PR. If you want to do this you should do it too for Schemas and Resources

**Review by umbreak — CHANGES_REQUESTED** [review]

File: tests/src/test/resources/application.conf [hunk 2]
```diff
@@ -1,6 +1,6 @@
 tests {
-    delta-uri = "http://localhost:8080/v1"
-    realm-uri = "http://"${"keycloak:8080"}"/auth/realms"
```
Comment (bogdanromanx) [inline tests/src/test/resources/application.conf:3]: this is required if you want to do local debugging

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

File: delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala (hunk 1 above)
Comment (imsdu) [reply delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala:183]: I did it because I had to declare the cache key in the serialization bindings

File: tests/src/test/resources/application.conf (hunk 2 above)
Comment (imsdu) [reply tests/src/test/resources/application.conf:3]: Resolver creation was failing because of this as it wasn't able to fetch the openId config

File: delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala (hunk 1 above)
Comment (bogdanromanx) [reply delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala:183]: this is required to be able to serialize the cache key

**Review by umbreak — APPROVED** [review]

File: tests/src/test/resources/application.conf (hunk 2 above)
Comment (bogdanromanx) [reply tests/src/test/resources/application.conf:3]: that's very strange; the docker compose plugin sets the `keycloak:8080` env variable.

Comment (imsdu) [reply tests/src/test/resources/application.conf:3]: Yep, but we use the realm-uri for realm payload for the config url so the `keycloak:8080` env variable resolves to localhost:xxxx which can't be reached by the delta instance as it is the address exposed outside the docker compose network. Inside it is `keycloak:8080`

**Review by bogdanromanx — APPROVED** [review]
